Test corpus failure (passes on dev):
```
~/go/src/github.com/cloudflare/bn256 $ tinygo test -tags="generic"
--- FAIL: TestG1Marshal (0.00s)
    bn256: malformed point
    FailNow is incomplete, requires runtime.Goexit()
--- FAIL: TestG2Marshal (0.00s)
    bn256: malformed point
    FailNow is incomplete, requires runtime.Goexit()
--- FAIL: TestBilinearity (0.12s)
    bad pairing result: bn256.GT(((68dce7c99407483c76cb1bf3bb03f68e774d8b4fbc9f0331666568fdd3f05e81, 769fc3874b83423cb70eb8b3944b1347b492f69a33c8019fa134ccb07c644eac), (236eba54528e5465382d345b8b902e6e931237c39b6b41562087a010977392eb, 8f06e7f42668c20b39fa5bdb9bf67956a3e53c9fd47ad2a8864d85ca2a82f16d), (895b2804b0f7df8fac0df1c4a7c72dc0027975ad4e9a922644a0aa2138207bf4, 353776a509896a6fd67d25d1371237f467d2842da9560c3f9be0cb82decd143a)),((750b5fa25d469acf70a24b36703cd2a2aecea8dfcb90c84ef616a54b23e9a30e, 2836aad94952d1cc78d5ac382095027e9f898805271f91f20b664f7a70ceeba3), (4adf959be15490d6e9f25fe5a9d0f0e901a9df3a8c0a1f8113095cfeb734c1f4, 05095ed820a5295f216ac4a15ab6384abcb7f703d0840275be00e7d06a3f4e2f), (7b5911518af9d6c2988faea27092dcab9cffeed106a9b29b2bb3e3278a73c45b, 3d22ec67ce9b09d03257bd687a27d2b3941b8b74f186f78660c4d57e7d459685)))
    FailNow is incomplete, requires runtime.Goexit()
    bad pairing result: bn256.GT(((405746fb787b3c1245b8dc989961764b957ea34ea298591c38a508cbe78edb7b, 4034e40a4e606d214519f41b95d7fd53ce26e706e460a92460aca52233debc1b), (6bc09600d659d0aeeda8ddc939327119bd691c416f5bdf01b3b4ac188c910f3e, 2443e00eeae6d91e0e1e57acfaf1787b5f6548e41471c3ef8574a6b5a1fe93d7), (009c12f056dea0dd26fbebb8bbff14934a77299de9acc194030c1a6b2bc91835, 07089f59fbc8593e18601cfbf395a1772acb45adcc954524bc978d965f51a4b4)),((59ffe78f6ecee0193ab2a39b5e69edf00aba4971c03132cf34540dfa7d1f4acf, 000c3b313fcfb1297dfd5fbe7435b77e49dde9f32372ea68fa77ebd7662139a1), (33ad0d7d30bf1d8b5dcc8015acc0f1ad92d0e3d69be2802abfdf73ee08405eb6, 745e794d081d2ce32b306163f17b87959be0b5aed103b2f3fd44258dd795165d), (62feb2204769f84c845a5f8b4d6a98c81df4156c8b8d2663f91676252ca61d16, 5b477ea70609b7f27c026ebfb76c3e4b7a569dbcba19c5e3d37bb4073f878aef)))
    FailNow is incomplete, requires runtime.Goexit()
--- FAIL: TestTripartiteDiffieHellman (0.09s)
    keys didn't agree
--- FAIL: TestDirtyUnmarshal (0.00s)
    bn256: malformed point
    FailNow is incomplete, requires runtime.Goexit()
--- FAIL: TestKnownHashes (0.01s)
    hash doesn't match a known value
    FailNow is incomplete, requires runtime.Goexit()
    hash doesn't match a known value
    FailNow is incomplete, requires runtime.Goexit()
    hash doesn't match a known value
    FailNow is incomplete, requires runtime.Goexit()
    hash doesn't match a known value
    FailNow is incomplete, requires runtime.Goexit()
    hash doesn't match a known value
    FailNow is incomplete, requires runtime.Goexit()
    hash doesn't match a known value
    FailNow is incomplete, requires runtime.Goexit()
    hash doesn't match a known value
    FailNow is incomplete, requires runtime.Goexit()
    hash doesn't match a known value
    FailNow is incomplete, requires runtime.Goexit()
    hash doesn't match a known value
    FailNow is incomplete, requires runtime.Goexit()
    hash doesn't match a known value
    FailNow is incomplete, requires runtime.Goexit()
    hash doesn't match a known value
    FailNow is incomplete, requires runtime.Goexit()
FAIL
FAIL	github.com/cloudflare/bn256	1.002s
```

--- a/interp/memory.go
+++ b/interp/memory.go
@@ -316,10 +316,14 @@ func (mv *memoryView) load(p pointerValue, size uint32) value {
 		panic("interp: load out of bounds")
 	}
 	v := obj.buffer.asRawValue(mv.r)
-	loadedValue := rawValue{
-		buf: v.buf[p.offset() : p.offset()+size],
+	loadedBuf := v.buf[p.offset() : p.offset()+size]
+	if _, writable := mv.objects[p.index()]; writable {
+		// This object's buffer is owned by this view, which means a later
+		// store may mutate it in place (see store below). Copy the loaded
+		// slice so the returned value is not aliased with the live buffer.
+		loadedBuf = append([]uint64(nil), loadedBuf...)
 	}
-	return loadedValue
+	return rawValue{buf: loadedBuf}
 }
 
 // Store to the value behind the given pointer. This overwrites the value in the
@@ -356,10 +360,6 @@ func (mv *memoryView) store(v value, p pointerValue) bool {
 		buffer := obj.buffer.asRawValue(mv.r)
 		obj.buffer = buffer
 		v := v.asRawValue(mv.r)
-		if writable {
-			// A partial load from this object may share the destination buffer.
-			v.buf = append([]uint64(nil), v.buf...)
-		}
 		for i := uint32(0); i < valueLen; i++ {
 			buffer.buf[p.offset()+i] = v.buf[i]
 		}

--- a/interp/testdata/store.ll
+++ b/interp/testdata/store.ll
@@ -4,11 +4,14 @@ target triple = "x86_64--linux"
 @overlap.buf = global [4 x i8] c"\01\02\03\04"
 @alias.src = global [4 x i8] c"\05\06\07\08"
 @alias.dst = global [2 x i8] zeroinitializer
+@reload.buf = global [4 x i8] c"\01\02\03\04"
+@reload.out = global [2 x i8] zeroinitializer
 
 define void @runtime.initAll() unnamed_addr {
 entry:
   call void @overlap.init(ptr undef)
   call void @alias.init(ptr undef)
+  call void @reload.init(ptr undef)
   ret void
 }
 
@@ -28,5 +31,19 @@ entry:
   %val = load i16, ptr %src
   store i16 %val, ptr @alias.dst
   store i8 9, ptr @alias.dst
+  ret void
+}
+
+define internal void @reload.init(ptr %context) unnamed_addr {
+entry:
+  ; First store makes reload.buf writable in the current memory view.
+  %tail = getelementptr [4 x i8], ptr @reload.buf, i32 0, i32 3
+  store i8 9, ptr %tail
+  ; Partial load whose result may share the underlying buffer.
+  %val = load i16, ptr @reload.buf
+  ; Subsequent in-place partial store; this must not corrupt %val.
+  store i8 99, ptr @reload.buf
+  ; Write the originally-loaded value to a separate global.
+  store i16 %val, ptr @reload.out
   ret void
 }

--- a/interp/testdata/store.out.ll
+++ b/interp/testdata/store.out.ll
@@ -5,7 +5,7 @@ target triple = "x86_64--linux"
 @alias.src = local_unnamed_addr global [4 x i8] c"\05\06\07\08"
 @alias.dst = local_unnamed_addr global [2 x i8] c"\09\07"
 @reload.buf = local_unnamed_addr global [4 x i8] c"c\02\03\09"
-@reload.out = local_unnamed_addr global [2 x i8] c"c\02"
+@reload.out = local_unnamed_addr global [2 x i8] c"\01\02"
 
 define void @runtime.initAll() unnamed_addr {
 entry:

--- a/interp/testdata/store.out.ll
+++ b/interp/testdata/store.out.ll
@@ -4,6 +4,8 @@ target triple = "x86_64--linux"
 @overlap.buf = local_unnamed_addr global [4 x i8] c"\01\01\02\09"
 @alias.src = local_unnamed_addr global [4 x i8] c"\05\06\07\08"
 @alias.dst = local_unnamed_addr global [2 x i8] c"\09\07"
+@reload.buf = local_unnamed_addr global [4 x i8] c"c\02\03\09"
+@reload.out = local_unnamed_addr global [2 x i8] c"c\02"
 
 define void @runtime.initAll() unnamed_addr {
 entry:

